### DISCUSSION TITLE
How to export a .ctf file after grain reconstruction

### DISCUSSION THREAD

**Discussion by umarmanan** [mtex-toolbox/mtex#1003] [Ask Anything]
hello Everyone,

I have .ctf format EBSD data of steel sample which has not indexed points as shown in the figure below. I want to reconstruct the grains and save them to a new file. 

For that i have used following command

ebsd.export('check1.ctf');

it saves a new .ctf file with "check1" name. the problem is when i open it again and want to save the grains using following command,

[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',10*degree);

it shows the following warnings and errors

Warning: Duplicate data points have been detected.
This may lead to inconsistencies in the point to Voronoi region relationship.
To avoid this behavior, call UNIQUE on the data points prior to calling VORONOIN. 
> In voronoin (line 74)
  In spatialDecomposition (line 35)
  In EBSD/calcGrains (line 45) 
Index exceeds array bounds.

Error in spatialDecomposition (line 39)
  D = D(1:size(X,1));

Error in EBSD/calcGrains (line 45)
[V,F,I_FD] = spatialDecomposition([ebsd.prop.x(:), ebsd.prop.y(:)],ebsd.unitCell,varargin{:});

Any help or clue will be highly appreciated. 
Thanks in Advance. 

![actual Condition](https://user-images.githubusercontent.com/78804260/116974701-2b4f0380-acc7-11eb-81b2-f27963b2a094.png)

**Comment by kilir** [comment]
Hi,
could you add on what you did to reconstruct the non-indexed points?
Cheers,
Rüdiger

**Reply by umarmanan** [reply]
Hi Rudiger,

Thanks for you response.

i am using following lines for ebsd data filling.

%Filling Data RAW code

ebsd=EBSD.load('EBSD Messungen Faisal NME B1 Feinmessung 2 Elementverteilungsdaten.ctf')
plotx2east

[grains,ebsd.grainId, ebsd.mis2mean]=ebsd.calcGrains;
ebsd(grains(grains.grainSize<=8),'notIndexed')=[]
[grains,ebsd.grainId ebsd.mis2mean]=ebsd.calcGrains;
figure
plot(ebsd)

% fill points
ebsdf = fill(ebsd,grains)
grod = calcGROD(ebsdf,grains);
[grains,ebsdf.grainId ebsdf.mis2mean]=ebsdf.calcGrains;
figure
plot(ebsdf); hold on
plot(grains.boundary); hold off


i want to export the file after this modification.
